src/intellisense/utils.rs

```diff
@@ -0,0 +1,104 @@
+use crate::intellisense::DxcCursor;
+use crate::intellisense::wrapper::DxcSourceOffsets;
+use crate::intellisense::ffi::DxcCursorFormatting;
+use winapi::shared::winerror::HRESULT;
+use winapi::shared::wtypes::BSTR;
+use winapi::shared::ntdef::LPSTR;
+use winapi::shared::ntdef::LPCSTR;
+use winapi::um::oleauto::{SysFreeString, SysStringLen, SysStringByteLen};
+
+
+pub(crate) fn from_bstr(string: BSTR) -> Result<String, HRESULT> {
```

Comment:
What's the reason for returning a HRESULT here? If it's not an error from the OS I think we should return something else.

Reply:
For `from_bstr` and `from_lpstr`, since they're consumed by the methods in wrapper.rs, a dummy error is needed. But I can change it to return Option<String> and handle the dummy HRESULT on the caller.

For the print utilities it was a lazy choice because they call the wrapper methods that themselves return HRESULT. But I'm unsure which error to return instead.